pkg/storage/ingest/reader.go

```diff
@@ -1043,6 +1084,12 @@ func newReaderMetrics(partitionID int32, reg prometheus.Registerer, bufferedReco
 		lastConsumedOffset:               lastConsumedOffset,
 		kprom:                            NewKafkaReaderClientMetrics(component, reg),
 	}
+
+	m.Service = services.NewTimerService(100*time.Millisecond, nil, func(context.Context) error {
```

Comment:
I can't see anyone starting/stopping this service.

Reply:
it's started here 
https://github.com/grafana/mimir/blob/314647e93954141427fb861dee9e9dda6007a0a3/pkg/storage/ingest/reader.go#L219-L228

and then stopped here
https://github.com/grafana/mimir/blob/314647e93954141427fb861dee9e9dda6007a0a3/pkg/storage/ingest/reader.go#L280-L285